lodash.js

```diff
@@ -277,7 +277,7 @@
   var reHasUnicode = RegExp('[' + rsZWJ + rsAstralRange  + rsComboRange + rsVarRange + ']');
 
   /** Used to detect strings that need a more robust regexp to match words. */
-  var reHasUnicodeWord = /[a-z][A-Z]|[A-Z]{2,}[a-z]|[0-9][a-zA-Z]|[a-zA-Z][0-9]|[^a-zA-Z0-9 ]/;
+  var reHasUnicodeWord = /[a-z][A-Z]|[A-Z]{2}[a-z]|[0-9][a-zA-Z]|[a-zA-Z][0-9]|[^a-zA-Z0-9 ]/;
 
```

Comment:
What if the word was `WINzip` _(3 upper case letters followed by a lower case)_.

Reply:
I tested on 4.17.0 without my changes and I get:
```javascript
_.words('WINzip') = ["WI","Nzip"]
_.words('WINžip') = ["WI","Nžip"]
```
And I tested with my changes and I get exactly the same results:
```javascript
_.words('WINzip') = ["WI","Nzip"]
_.words('WINžip') = ["WI","Nžip"]
```
This is because the problematic RegEx is only used by `hasUnicodeWord` to **_test_** if we should use `unicodeWords` or `asciiWords` for the **_actual_** parsing:

```javascript
    function words(string, pattern, guard) {
      string = toString(string);
      pattern = guard ? undefined : pattern;

      if (pattern === undefined) {
        return hasUnicodeWord(string) ? unicodeWords(string) : asciiWords(string);
      }
      return string.match(pattern) || [];
    }
```

Do you want me to add this particular test case? I think this _pattern_ it's already covered by `test.js`:

```javascript
    QUnit.test('should work with compound words', function(assert) {
      assert.expect(12);

      assert.deepEqual(_.words('12Feet'), ['12', 'Feet']);
      assert.deepEqual(_.words('aeiouAreVowels'), ['aeiou', 'Are', 'Vowels']);
      assert.deepEqual(_.words('enable 6h format'), ['enable', '6', 'h', 'format']);
      assert.deepEqual(_.words('enable 24H format'), ['enable', '24', 'H', 'format']);
      assert.deepEqual(_.words('isISO8601'), ['is', 'ISO', '8601']);
      assert.deepEqual(_.words('LETTERSAeiouAreVowels'), ['LETTERS', 'Aeiou', 'Are', 'Vowels']);
      assert.deepEqual(_.words('tooLegit2Quit'), ['too', 'Legit', '2', 'Quit']);
      assert.deepEqual(_.words('walk500Miles'), ['walk', '500', 'Miles']);
      assert.deepEqual(_.words('xhr2Request'), ['xhr', '2', 'Request']);
      assert.deepEqual(_.words('XMLHttp'), ['XML', 'Http']);
      assert.deepEqual(_.words('XmlHTTP'), ['Xml', 'HTTP']);
      assert.deepEqual(_.words('XmlHttp'), ['Xml', 'Http']);

      console.log(`_.words('WINzip') = ${JSON.stringify(_.words('WINzip'))}`);
    });

    QUnit.test('should work with compound words containing diacritical marks', function(assert) {
      assert.expect(3);

      assert.deepEqual(_.words('LETTERSÆiouAreVowels'), ['LETTERS', 'Æiou', 'Are', 'Vowels']);
      assert.deepEqual(_.words('æiouAreVowels'), ['æiou', 'Are', 'Vowels']);
      assert.deepEqual(_.words('æiou2Consonants'), ['æiou', '2', 'Consonants']);

      console.log(`_.words('WINžip') = ${JSON.stringify(_.words('WINžip'))}`);
    });
```